app/js/workspace/workspaceSettingsService.js

```diff
@@ -109,8 +156,35 @@ define(['angular', 'lodash'], function(angular, _) {
     }
 
     function isValueView() {
-      return workspaceSettings.effectsDisplay === 'smaa' ||
-        workspaceSettings.effectsDisplay === 'deterministicMCDA';
+      return workspaceSettings.displayMode === 'values';
+    }
+
+    function getWarnings(settings) {
+      var warnings = [];
+      if (hasNoEnteredData(settings)) {
+        warnings.push('No entered data available.');
+      } else if (hasNoEnteredEffect(settings)) {
+        warnings.push('No entered data available for deterministic analysis.');
+      } else if (hasNoEnteredDistribution(settings)) {
+        warnings.push('No entered data available for SMAA analysis.');
+      }
```

Comment:
if every push is in an else branch you can just return the warning directly

Reply:
True, but I wanted it to follow the same style as we do in other places. Also this way more warnings can be added easily if ever needed.

Reply:
IMO it's distracting to use this format if it's not necessary, I kept thinking I was missing something :) context-free code!